lib/fileutils.c

```diff
@@ -265,3 +266,28 @@ int ul_copy_file(int from, int to)
 #endif
 	return 0;
 }
+
+/* Copies the contents of a file. Returns -1 on read error, -2 on write error. */
```

Comment:
I think it would be good to add UL_READ_ERROR and UL_WRITE_ERROR definitions to include/fileutils.h that define these negative numbers.